sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala

```diff
@@ -133,7 +133,7 @@ class MicroBatchExecution(
   // Store checkpointIDs for state store checkpoints to be committed or have been committed to
   // the commit log.
   // operatorID -> (partitionID -> array of uniqueID)
-  private val currentStateStoreCkptId = MutableMap[Long, Array[Array[String]]]()
+  private var currentStateStoreCkptId: Option[MutableMap[Long, Array[Array[String]]]] = None
```

Comment:
I don't think the option changes here are necessary - it's only important for the serialization case class, not the rest imho. @siying please let me know if you disagree

Reply:
@brkyvz I agree. Only the one in commit log is the one I am worried about.